### PR TITLE
[query] Switch _buildEncoder methods to use CodeBuilder infrastructure

### PR DESCRIPTION
**Pull request by chrisvittal**
This streamlines the logic significantly. No attempt has been made to have the encoder methods take
PCode/EmitCode directly, but that is the future direction here.

### CODE REVIEW

**Comment by chrisvittal** [comment]
Ok, hopefully ironed out the bugs.

**Comment by chrisvittal** [comment]
@tpoterba The tests now all pass. I'd appreciate your input on 32965c8 , the last commit in this sequence. It shortens the names of methods, possibly truncating them more than we already do, but I had found that something was blowing memory related to large classes/constants and once the types are much larger than a few KB, they're not really that useful.

Thanks!

**Comment by tpoterba** [comment]
yes, truncating more is a great idea.

**Comment by chrisvittal** [comment]
This is ready for review or merging.

File: hail/src/main/scala/is/hail/types/encoded/ETransposedArrayOfStructs.scala [hunk 1]
```diff
@@ -212,76 +212,55 @@ final case class ETransposedArrayOfStructs(
...
-
-      val encoders = fieldGroup.map { encodedField =>
-        val fidx = elementPStruct.fieldIdx(encodedField.name)
-        val pf = elementPStruct.fields(fidx)
-        val encodeField = encodedField.typ.buildEncoder(pf.typ, groupMB.ecb)
-        val elem = () => arrayPType.elementOffset(addr, len, j)
+      if (!ef.typ.required) {
+        cb.forLoop(cb.assign(j, 0), j < len, cb.assign(j, j + 1), {
+          array.loadElement(cb, j).consume(cb, { /* do nothing */ }, { pbsc =>
+            // FIXME may be bad if structs get memoized into their fields, otherwise
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/types/encoded/ETransposedArrayOfStructs.scala:241]: I think this should probably be fine, good to note though.

**Pull request merged**